src/core/src/package_managers/AptitudePackageManager.py

```diff
@@ -524,3 +531,56 @@ def update_os_patch_configuration_sub_setting(self, patch_configuration_sub_sett
     def do_processes_require_restart(self):
         """Defaulting this for Apt"""
         return False
+
+    def is_reboot_pending(self):
+        """ Checks if there is a pending reboot on the machine. """
+        ubuntu_pro_client_check_success = False
+        ubuntu_pro_client_reboot_status = False
+        reported_reboot_status = False
+        default_exception = None
+        default_pending_file_exists = False
+        default_pending_processes_exists = False
+
+        # Default reboot check.
+        try:
+            default_pending_file_exists = os.path.isfile(self.REBOOT_PENDING_FILE_PATH)
+            default_pending_processes_exists = self.do_processes_require_restart()
+        except Exception as error:
+            default_exception = repr(error)
```

Comment:
Do we need to log?

Reply:
we log every thing at the end of the function as per our last discussion on single log/ single return per method.

Reply:
Looks good. Closing.